sync3/handler/connstate_live.go

```diff
@@ -330,6 +330,15 @@ func (s *connStateLive) processLiveUpdateForList(
 ) (hasUpdates bool) {
 	switch update := up.(type) {
 	case *caches.RoomEventUpdate:
+		if update.EventData.StateKey == nil && s.userCache.ShouldIgnore(update.EventData.Sender) {
```

Comment:
This feels like the wrong place for this check. If you have no lists for example and instead are using a direct room sub then surely we'd not do this check?

Reply:
Oh true enough. Tbh I just found spotted the log line in the test output and didn't think too hard about whether this was the right place. I'll move it up to processLiveUpdate.